SonarQube.Common/Serializer.cs

```diff
@@ -43,9 +43,11 @@ public static void SaveModel<T>(T model, string fileName)
             settings.NamespaceHandling = NamespaceHandling.OmitDuplicates;
             settings.OmitXmlDeclaration = false;
 
-            using (XmlWriter writer = XmlWriter.Create(fileName, settings))
+            using (MemoryStream stream = new MemoryStream())
+            using (XmlWriter writer = XmlWriter.Create(stream, settings))
             {
                 serializer.Serialize(writer, model);
```

Comment:
The first time the XmlSerializer is called it reflects over the type being serialized and generates serialization code which is relatively slow. This change serialises the data to memory first then writes the data to a file, rather than holding the file open while the type is serialized.